src/dotnet/APIView/APIViewWeb/Models/RenderedCodeFile.cs

```diff
@@ -43,15 +50,83 @@ internal CodeLine[] RenderText(bool skipDiff = false)
         {
             if (skipDiff)
             {
-                return CodeFileRenderer.Instance.Render(CodeFile, enableSkipDiff: skipDiff);
+                RenderResult = CodeFileRenderer.Instance.Render(CodeFile, enableSkipDiff: skipDiff);
+                return RenderResult.CodeLines;
             }
 
             if (_renderedText == null)
             {
-                _renderedText = CodeFileRenderer.Instance.Render(CodeFile);
+                RenderResult = CodeFileRenderer.Instance.Render(CodeFile);
+                _renderedText = RenderResult.CodeLines;
             }
 
             return _renderedText;
         }
+
+        public CodeLine[] GetCodeLineSection(int sectionId)
+        {
+            var section = RenderResult.Sections[sectionId];
+            var result = new List<CodeLine>();
+
+            using (IEnumerator<TreeNode<CodeLine>> enumerator = section.GetEnumerator())
+            {
+                enumerator.MoveNext();
```

Comment:
Do we need this MoveNext? First iteration is called with 2 MoveNext()?

Reply:
Yes the first item is a placeholder that does not need to be displayed.